File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }

        unordered_map<char, int> freq;

        for (char c : s) {
            freq[c]++;
        }

        for (char c : t) {
            auto it = freq.find(c);

            if (it == freq.end()) {
                return false;
            }

            it->second--;

            if (it->second == 0) {
                freq.erase(it);
            }
        }

        return freq.empty();
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        freq = [0] * 26
        base = ord('a')

        for ch in s:
            freq[ord(ch) - base] += 1
        for ch in t:
            idx = ord(ch) - base
            freq[idx] -= 1
            if freq[idx] < 0:
                return False

        return True


        